purchases/index.js

```diff
@@ -511,12 +511,18 @@ function listPurchases(db, options, callback) {
         callback = options;
         options = {};
     }
-    db.query(`SELECT * FROM purchase ORDER BY started_at DESC ${parseLimit(options)} ${parseOffset(options)}`, (error, result) => {
-        if(error) {
-            console.error("Failed to get purchases to list", error);
-            return callback(error);
+    db.query(`SELECT * FROM purchase ORDER BY started_at DESC ${parseLimit(options)} ${parseOffset(options)}`, (limitedError, limitedResult) => {
+        if(limitedError) {
```

Comment:
Don't rename `error` - it is convention to use that name for every error, and it is okay to overload the name in later calls.

Same for the result honestly. `limitedResult` means very little to me. If you want a different name, it should denote the purpose. The first query is for results, the second is for a count of results. Use that in the naming.